src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -13,9 +13,39 @@ internal partial class ClrThreadPool
         /// </summary>
         private partial class HillClimbing
         {
-            // Config values pulled from CoreCLR
-            // TODO: Move to runtime configuration variables.
-            public static HillClimbing ThreadPoolHillClimber { get; } = new HillClimbing(4, 20, 100 / 100.0, 8, 15 / 100.0, 300 / 100.0, 4, 20, 10, 200, 1 / 100.0, 200 / 100.0, 15 / 100.0);
+            private static Lazy<HillClimbing> s_threadPoolHillClimber = new Lazy<HillClimbing>(CreateHillClimber, true);
+            public static HillClimbing ThreadPoolHillClimber => s_threadPoolHillClimber.Value;
+
+            private static HillClimbing CreateHillClimber()
+            {
+                // Default values pulled from CoreCLR
+                int wavePeriod = (int?)AppContext.GetData("HillClimbing_WavePeriod") ?? 4;
```

Comment:
These values may show up as string (when set from config file). You may want to write a helper function that deals with strings; and that ignores unexpected types.